nototools/check_familyname_and_styles.py

```diff
@@ -0,0 +1,174 @@
+
+You can check the file, and generate a list of font file names from it.
+This list can be passed to noto_names to generate the name data.
+
+The file is a list of Noto family names (see noto_fonts.py) interspersed with
+definitions of what WWS style combinations apply to that file.  See
+_get_stylenames() for the format.  Each style definition applies to each
+following family until the next style definition."""
+
+import argparse
+import re
+
+from nototools import noto_fonts
+
+_style_re = re.compile(r'--\s+(.*)\s+--')
+_extended_style_re = re.compile(r'^([TRBH]+)(?:/([CR]+)(?:/([RI]+))?)?$')
+
+# Below we use the longest names we intend, so that the noto_names code can
+# identify which families need extra short abbreviations.  The style of
+# abbreviation is based on the longest names in the family.
+
+_WEIGHT_NAMES = {
+    'T': 'Thin',
+    'R': 'Regular',
+    'B': 'Bold',
+    'H': 'ExtraBold' # Nee 'Heavy'. Not 'Black' because 'ExtraBold' is longer.
+}
+
+_WIDTH_NAMES = {
+    'C': 'SemiCondensed', # We use this since it is longer. We don't expect to
+                          # use ExtraCondensed.
+    'R': ''
+}
+
+_ITALIC_NAMES = {
+    'I': 'Italic',
+    'R': '',
+}
+
+def _get_stylenames(styles):
+  """Returns the list of style names for the encoded styles.  These are the
+  (master-ish) style names encoded as weights / widths/ italic, where weights
+  is any of 'T', 'R', 'B', or 'H', widths any of 'C' or 'R', and italic 'I'.
+  If there's not an italic then the italic is omitted, if there's only
+  regular width and no italic then widths are omitted."""
+  m = _extended_style_re.match(styles)
+  assert m
+  weights = m.group(1)
+  widths = m.group(2) or 'R'
+  slopes = m.group(3) or 'R'
+  # print '%s: %s, %s, %s' % (styles, weights, widths, slopes)
+  names = []
+  for wd in widths:
+    width_name = _WIDTH_NAMES[wd]
+    for wt in weights:
+      weight_name = _WEIGHT_NAMES[wt]
+      for it in slopes:
+        italic_name = _ITALIC_NAMES[it]
+        final_weight_name = weight_name
+        if wt == 'R' and (width_name or italic_name):
+          final_weight_name = ''
+        names.append(width_name + final_weight_name + italic_name)
+  return names
+
+
+def check_familyname(name, styles):
+  notofont = noto_fonts.get_noto_font('unhinted/' + name + '-Regular.ttf')
+  if not notofont:
+    print 'Error: could not parse', name
+    return False
+  print name, noto_fonts.noto_font_to_wws_family_id(notofont), styles
+  return True
+
+
+def generate_family_filenames(name, styles):
+  """Name is the family name portion of a Noto filename.  Styles is the
+  encoding of the styles, see _get_stylenames."""
+  stylenames = _get_stylenames(styles)
+  return [name + '-' + s + '.ttf' for s in stylenames]
+
+
+def _for_all_familynames(namefile, fn):
+  """Call fn passing the family name and style descriptor for
+  all families in namefile. '#' is a comment to eol, blank lines are
+  ignored."""
+  styles = 'RB'
```

Comment:
Well, it  was either setting the initial value, or failing with an error if it is not set before the first filename.  I guess it should be explicitly set, you're right.